### ISSUE TITLE
Should MA0046 allow the use of EventHandler<TSender,TEventArgs>?

### ISSUE THREAD

**Issue by girlpunk** [meziantou/Meziantou.Analyzer#1434]
### Version of the Meziantou.Analyzer NuGet package

3.0.124

### Rule Identifier

MA0046

### Target Framework

net10.0

### C# Language version

14.0

### .NET SDK version

10.0.111

### Description

Not 100% if this is a bug or I'm misunderstanding Microsoft's documentation, but should MA0046 allow the use of `EventHandler<TSender,TEventArgs>`?

### Reproduction Steps

Minimal code:

```c#
public event EventHandler<MyEventSource, MyEventArgs>? UpdateEvent;
```

### Other information

https://learn.microsoft.com/en-us/dotnet/api/system.eventhandler-2?view=net-10.0
